### ISSUE TITLE
Error: table index is nil

### ISSUE THREAD

**Issue by vinceallenvince** [karpathy/char-rnn#83]
I can train and sample files just fine on OS X. However, on a machine running Ubuntu 14.04, when trying to sample (ie. th sample.lua...), I get the following error. I've tried installing Torch w both luajit and plain lua (ie. TORCH_LUA_VERSION=LUA51 ./install.sh). But I still get the error. Any ideas? Thx.

/root/torch/install/bin/lua: /root/torch/install/share/lua/5.1/torch/File.lua:283: table index is nil
stack traceback:
	/root/torch/install/share/lua/5.1/torch/File.lua:283: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:268: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:268: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:268: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	...
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:268: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:282: in function 'readObject'
	/root/torch/install/share/lua/5.1/torch/File.lua:305: in function 'load'
	sample.lua:67: in main chunk
	[C]: in function 'dofile'
	.../torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
	[C]: ?

**Comment by hughperkins** [comment]
Not saying it will fix the problem, but seems like you are using a fairly old version of char-rnn?

**Comment by EricZeiberg** [comment]
How can you tell?

**Comment by hughperkins** [comment]
becuaase line 67 of karpathy master doesnt call `File:load` :-P

**Comment by vinceallenvince** [comment]
Thx, pulled the latest but still get the error on another line.
sample.lua:86: in main chunk

**Comment by hughperkins** [comment]
Ok.  Are you sampling from a cv file created using the same version of char-rnn, or a different version?

**Comment by vinceallenvince** [comment]
I'm sampling from a file created using an earlier version. I'll train a file with the new version and sample and see what happens.

**Comment by vinceallenvince** [comment]
Wow, ok.. I trained until I got one checkpoint file and sampled it and it worked! Thx.

**Comment by EricZeiberg** [comment]
If the issue has been fixed, please close this.

**Comment by vinceallenvince** [comment]
One more question.. I notice if I'm on my Ubuntu machine and try to sample checkpoints trained on OS X, I get the error. Is that expected? In other words, i should only try to sample from checkpoints trained on the same environment? Thx.

**Comment by hughperkins** [comment]
Are you using the same versoin on each?

**Comment by vinceallenvince** [comment]
Yes, just tested training on OS X and sampling a checkpoint from the training on Ubuntu. Same error.

**Comment by hughperkins** [comment]
Interesting.  I suppose it is to do with endian-ness plausibly.  Does your Mac have an intel x86 processor, or some kind of 68000 type non-Intel processor?

**Comment by soumith** [comment]
if you want to have cross-platform checkpoints, torch.save in ascii mode. torch.save('checkpoint.t7', net, 'ascii')

**Comment by soumith** [comment]
if you do 64-bit OSX and 64-bit Ubuntu, both on x64 processors. they should be fully compatible in terms of serialization. they only start getting weird when the processor endianness changes:

https://github.com/torch/torch7/issues/5

**Comment by EricZeiberg** [comment]
Also, be wary of saving the checkpoint under the same name every time, iirc
it will override the previous one.
